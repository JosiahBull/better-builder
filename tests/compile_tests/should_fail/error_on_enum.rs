use better_builder::BetterBuilder;

#[derive(BetterBuilder)]
enum Cart {
    Owner(String),
    NumWheels(u8),
    NumSeats(u8),
    Inventory(Vec<String>),
}

fn main() {}
