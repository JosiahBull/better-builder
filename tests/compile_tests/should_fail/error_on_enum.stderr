error: BetterBuilder can only be derived on structs.
 --> tests/compile_tests/should_fail/error_on_enum.rs:4:1
  |
4 | / enum Cart {
5 | |     Owner(String),
6 | |     NumWheels(u8),
7 | |     NumSeats(u8),
8 | |     Inventory(Vec<String>),
9 | | }
  | |_^
